### ISSUE TITLE
YAML alias usage is brittle

### ISSUE THREAD

**Issue by hmijail** [ThrowTheSwitch/Ceedling#74]
I don't know if this is Ceedling's or the YAML library's problem.

YAML allows usage of aliases: define something at one point, and then refer to it later in the same file:

```
:flags:
  :test:
    :common: &common_flags
      - -g      
    :compile:
      :*:
        - *common_flags
        - -Wall 
    :link:
      :*:
        - *common_flags
        - -L/opt/local/lib/
```

This works, and I think is quite useful. But: 
- if the :common: section happens to be empty, the rest of sections (:compile: and :link : ) seem to get ignored. 
  This can be mitigated by adding an empty item:

```
:flags:
  :test:
    :common: &common_flags
      -
    :compile:
      ...
```
- However, if an empty item appears at any position in the :common: list, all of the :common: items get ignored

So, both problems together mean that every time you change an item in the list you should check that the list stays either with ONLY an empty item, OR with some item and NO empty items. Which is prone to error if you tend to selectively comment out things in an existing long list of flags.

If this is some bug in Ceedling's handling of the output of the YAML parser, then it would be nice to have it fixed. 
However, if this is rather a bug in the YAML parser, then sorry for the noise :)

**Comment by mvandervoord** [comment]
Sadly this is just the way YAML parses data. We don't have any control over this. The correct way to set that to nothing is to go like this:

```
:flags:
  :test:
    :common: &common_flags []
```

That tells YAML that it's an empty array.

**Comment by mvandervoord** [comment]
Whoops. Prematurely closed. This should be made more clear in the documentation.

**Comment by mkarlesky** [comment]
Yes, this is primarily a YAML issue, but we can be more helpful in the documentation. I just added [a subsection to the YAML page in the documentation site](https://throwtheswitch.github.io/Ceedling/latest/configuration/project-file/?h=yaml#some-yaml-learnin) that briefly discusses these issues and links to more thorough resources. It also highlights that `ceedling dumpconfig` provides clarity on the resulting hash created from any YAML aliases and Ceedling mixins.
